compiler-rt/lib/radsan/radsan_user_interface.cpp

```diff
@@ -32,23 +32,21 @@ std::function<OnErrorAction()> createErrorActionGetter() {
       return OnErrorAction::Continue;
   };
 
-  auto user_mode = __sanitizer::GetEnv("RADSAN_ERROR_MODE");
-  if (user_mode == nullptr) {
-    return exit_getter;
-  }
+  CHECK(radsan::IsInitialized());
+  const char* user_mode = radsan::flags()->error_mode;
 
   if (std::strcmp(user_mode, "interactive") == 0) {
     return interactive_getter;
   }
-
-  if (std::strcmp(user_mode, "continue") == 0) {
+  else if (std::strcmp(user_mode, "continue") == 0) {
     return continue_getter;
   }
-
-  if (std::strcmp(user_mode, "exit") == 0) {
+  else if (std::strcmp(user_mode, "exit") == 0) {
     return exit_getter;
   }
 
+  __sanitizer::Printf("WARNING Invalid error mode: %s. Assuming 'exit'\n", user_mode);
```

Comment:
What made you choose to use the `__sanitizer` printf here instead of a standard one?

Reply:
Matching from other sanitizers (check out msan_report.cpp, for instance). Similar to using the atomics that are defined internally, I believe it's to run on systems that don't have standard headers

Reply:
See the other comment about recursion hell.

If you intercept a function then call a method like `std::printf` and that also calls a method that could be intercepted, you go into a recursive loop you can't recover from. I think that is more of the reason why they don't use standard implementations, but roll their own.